Methane_MarkDown.Rmd

```diff
@@ -111,3 +110,206 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
+  mutate(new_col= rows)
+colnames(data)<- c("PREINDUSTRIAL_CH4", "NATURAL_CH4", "LIFETIME_SOIL", "LIFETIME_STRAT", "run_number")
+```
+
+Using a `for` loop function, we can run both parameters of the data frame- read as different columns by our code. 
+
+```{r function_mod}
+multi_param<- function(core, pdata){ #create input for function
+  for(p in colnames(pdata)) { 
+    var<- do.call(p, list()) #create vector for row running
+    old_value<- fetchvars(core, NA, var) #old variable called to be changed
+    unit<- as.character(old_value[["units"]]) 
+    setvar(core, NA, var, pdata[p][[1]], unit) #set row vars from old to new values
+    reset(core) #reset core to run new values
+    run(core) #run new params
+    result<- fetchvars(core, 2000:2100) #results for x years
+  }
+  result
+}
+```
+
+To run our values through this new function, we create another `for ` loop to put in each row of data one by one. Then with `bind_rows` we can use these created values to form a data frame.
+
+```{r output}
+output<- list() #make an empty list
+for(rownum in 1:nrow(data)){ #loop starting with 1st row of data going from 1-end of data
+  output[[rownum]]<- multi_param(core, data[rownum,][-5]) #output as a row for inputs of all columns in row running(1-end data) besides 5th
+  output[[rownum]]$run_number<- data$run_number[[rownum]] #now that row has finished without 5th row, put 5th row back in as guide for next row run
+}
+
+output_data<- output %>% bind_rows() #create a data frame with output
+full_data<- left_join(output_data, data, by= "run_number") #add in param values columns to results of loop df by pairing with row_number column
+```
+
+### Graphing multiple parameters
+
+Now that we've created multi-parameter data, we can visualize our results. Using the `pivot_longer` function from `tidyr` we first group our parameters into two columns: `parameter` and `param_value`. This will make graphing much easier, since we are now only working with one variable type.
+
+```{r piv-long-data}
+long_fdata<- full_data %>%
+  pivot_longer(!c(scenario, year, variable, value, units, run_number), names_to= "parameter", values_to= "param_value")
+```
+
+To visualize the spread of our new values we can create a histogram broken into four windows with `facet_wrap`.
+
+```{r long-hist}
+long_fdata %>%
+  ggplot(aes(x= param_value, fill= parameter))+
+  geom_histogram(alpha= 0.6, binwidth= 7)  +
+   facet_wrap(~parameter, scales = "free_y")
+```
+To show the differences in values between each run, we can create a bar chart grouping parameter by `run_number`. Although the values for each run stay consistent, we must use the `summarise` and `mean` functions to condense our data frame into single values before graphing. 
+
+```{r bar_params}
+mean_fdata<- full_data %>%
+  group_by(run_number) %>%
+  summarise(meanNatM_TgCH4= mean(NATURAL_CH4), meanPreM_ppbvCH4= mean(PREINDUSTRIAL_CH4), meanSoilM_Years= mean(LIFETIME_SOIL), meanStratM_Years= mean(LIFETIME_STRAT))
```

Comment:
These names are a bit of a mouthful. You can use a string as a new name for the summarize function, like `summarize("Mean of natural emissions" = mean(NATURAL_CH4))` or something. Just a suggestion, you can keep as is if it makes sense to you!

You could also call them whatever (`mean1 = mean(...), mean 2 = mean(...)`) and then rename them using the `scale_fill_manual()` function in your ggplot call.